ui/pages/confirmations/hooks/gas/useIsGaslessLoading.test.ts

```diff
@@ -0,0 +1,125 @@
+import { renderHook } from '@testing-library/react-hooks';
+import { useSelector } from 'react-redux';
+import { useConfirmContext } from '../../context/confirm';
+import { getUseTransactionSimulations } from '../../../../selectors';
+import { useIsInsufficientBalance } from '../useIsInsufficientBalance';
+import { useIsGaslessSupported } from './useIsGaslessSupported';
+import { useIsGaslessLoading } from './useIsGaslessLoading';
+
+jest.mock('../../context/confirm');
+jest.mock('./useIsGaslessSupported');
+jest.mock('../useIsInsufficientBalance');
+jest.mock('react-redux', () => ({
+  ...jest.requireActual('react-redux'),
+  useSelector: jest.fn(),
+}));
+
+const mockedUseConfirmContext = jest.mocked(useConfirmContext);
+const mockedUseIsGaslessSupported = jest.mocked(useIsGaslessSupported);
+const mockedUseIsInsufficientBalance = jest.mocked(useIsInsufficientBalance);
+const mockedUseSelector = useSelector as jest.Mock;
+
+describe('useIsGaslessLoading', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  const setup = ({
+    simulationEnabled,
+    gaslessSupported,
+    insufficientBalance,
+    gasFeeTokens,
+  }: {
+    simulationEnabled: boolean;
+    gaslessSupported: boolean;
+    insufficientBalance: boolean;
+    gasFeeTokens?: object | null;
+  }) => {
+    mockedUseSelector.mockImplementation((selector) => {
```

Comment:
We always try and avoid mocking selectors in the unit tests to make the tests less brittle, as recommended by Redux.

We don't even need to mock the confirm context as we have the `renderHookWithConfirmContextProvider` helper that creates a mock store and transaction using real state.

Reply:
Removed mocked selectors.